controllers/user_reconciler.go

```diff
@@ -0,0 +1,298 @@
+			out, err := exec.Command("/bin/bash", "-c", updateCommad).Output()
+			if err != nil {
+				log.Errorf("error %s", err)
+			}
+			userpwd := fmt.Sprint(strings.TrimSpace(string(out)), "\n")
+			htpasswds = append(htpasswds, []byte(userpwd)...)
+		}
+	}
+
+	htpasswdSecret := openshiftuser.NewHTPasswdSecret(workshop, r.Scheme, HTPASSWD_SECRET_NAME, HTPASSWD_SECRET_NAMESPACE_NAME, htpasswds)
+	if err := r.Create(context.TODO(), htpasswdSecret); err != nil && !errors.IsAlreadyExists(err) {
+		return reconcile.Result{}, err
+	} else if err == nil {
+		log.Infof("Created  %s secret", htpasswdSecret.Name)
+	} else if errors.IsAlreadyExists(err) {
+		// Get secret
+		secretFound := &corev1.Secret{}
+		if err := r.Get(context.TODO(), types.NamespacedName{Name: HTPASSWD_SECRET_NAME, Namespace: HTPASSWD_SECRET_NAMESPACE_NAME}, secretFound); err == nil {
+			for _, secretData := range secretFound.Data {
+				encodedSecret := base64.StdEncoding.EncodeToString(secretData)
+				decodeSecret, err := base64.StdEncoding.DecodeString(encodedSecret)
+				if err != nil {
+					log.Errorf("Error %s", err)
+				}
+				log.Infoln(string(decodeSecret))
+				countUsers = strings.Count(string(decodeSecret), userPrefix)
+			}
+			if totalUsers > countUsers || totalUsers < countUsers {
+				if err := r.Delete(context.TODO(), htpasswdSecret); err != nil {
+					return reconcile.Result{}, err
+				}
+				if err := r.Create(context.TODO(), htpasswdSecret); err != nil && !errors.IsAlreadyExists(err) {
+					return reconcile.Result{}, err
+				} else if err == nil {
+					log.Infof("Created  %s secret", htpasswdSecret.Name)
+				}
+			}
+		}
+	}
+
+	return reconcile.Result{}, nil
+}
+
+// deleteUsers delete users in openshift cluster
+func (r *WorkshopReconciler) deleteUsers(workshop *workshopv1.Workshop) (reconcile.Result, error) {
+	var userList []string
+	userPrefix := workshop.Spec.UserDetails.UserNamePrefix
+	labelSelector, err := labels.Parse(UserLabelSelector)
+	if err != nil {
+		log.Errorf("Error %s", err)
+	}
+
+	ListUsers := &userv1.UserList{}
+	listOps := &client.ListOptions{
+		LabelSelector: labelSelector,
+	}
+	if err := r.List(context.TODO(), ListUsers, listOps); err != nil {
+		log.Errorf("Error %s", err)
+	}
+	for _, user := range ListUsers.Items {
+		username := user.Name
+		userList = append(userList, username)
+	}
+
+	createdUsers := len(userList)
+	for createdUsers >= 1 {
+		username := fmt.Sprint(userPrefix, createdUsers)
+		if result, err := r.deleteOpenshiftUser(workshop, r.Scheme, username); util.IsRequeued(result, err) {
+			return result, err
+		}
+		createdUsers--
+	}
+	if result, err := r.DeleteUserHtpasswd(workshop); err != nil {
+		return result, err
+	}
+
+	//Success
+	return reconcile.Result{}, nil
+}
+
+// deleteOpenshiftUser delete OpenShift user
+func (r *WorkshopReconciler) deleteOpenshiftUser(workshop *workshopv1.Workshop, scheme *runtime.Scheme, username string) (reconcile.Result, error) {
```

Comment:
@MaazMS why not just call it `deleteUser`

Reply:
deleteOpenshiftUser is delete users in openshift,
`deleteUser` is better name of this method.

Reply:
`deleteUsers` method used in workshop_controller.go  and inside `deleteUsers` we call two methods. 
1. `deleteOpenshiftUser` method to delete users. 
2. `DeleteUserHtpasswd` method to delete htpasswd . 

To avoid similar name for two method i used `deleteOpenshiftUser` method name for delete user.